pkg/services/accesscontrol/acimpl/accesscontrol.go

```diff
@@ -226,7 +230,31 @@ func (a *AccessControl) Check(ctx context.Context, req accesscontrol.CheckReques
 		Relation: req.Relation,
 		Object:   req.Object,
 	}
-	in := &openfgav1.CheckRequest{TupleKey: key}
+
+	in := &openfgav1.CheckRequest{
+		TupleKey: key,
+	}
+
+	// Add contextual tuple with parent folder
+	if req.Parent != "" {
+		ns, err := claims.ParseNamespace(req.Namespace)
+		if err != nil {
+			return false, err
+		}
+
+		contextualTuple := &openfgav1.TupleKey{
+			User:     zanzana.NewScopedTupleEntry(zanzana.TypeFolder, req.Parent, "", strconv.FormatInt(ns.OrgID, 10)),
+			Relation: zanzana.RelationParent,
+			Object:   req.Object,
+		}
+
+		in.ContextualTuples = &openfgav1.ContextualTupleKeys{
+			TupleKeys: []*openfgav1.TupleKey{
+				contextualTuple,
+			},
+		}
+	}
+
```

Comment:
I would imagine living in the zanzana client https://github.com/grafana/grafana/blob/main/pkg/services/authz/zanzana/client/client.go#L88-L95.

Right now we are exposing zanzana only methods through the access control interface, I guess this was done so we did not have to inject the zanzana client into dashboard service?

We should try to aling our client interface for zanzana with https://github.com/grafana/authlib/blob/main/authz/multitenant_client.go#L48-L50 and some extra methods